protected/humhub/modules/file/models/File.php

```diff
@@ -327,48 +389,108 @@ public function setStoredFile($file, $skipHistoryEntry = false)
      * @since 1.10
      */
     public function setStoredFileContent($content, $skipHistoryEntry = false)
     {
         $this->beforeNewStoredFile($skipHistoryEntry);
-        $this->store->setContent($content);
+        $this->getStore()->setContent($content);
         $this->afterNewStoredFile();
     }
 
     /**
      * Steps that must be executed before a new file content is set.
+     *
      * @param bool $skipHistoryEntry Skipping the creation of a history entry, even if enabled by the record
+     *
+     * @throws Exception|IntegrityException|InvalidConfigException
      */
     private function beforeNewStoredFile(bool $skipHistoryEntry)
     {
         if ($this->isNewRecord) {
             throw new Exception('File Record must be saved before setting a new file content.');
         }
 
-        if ($this->store->has() && FileHistory::isEnabled($this) && !$skipHistoryEntry) {
+        $store = $this->getStore();
+
+        if ($store->has() && FileHistory::isEnabled($this) && !$skipHistoryEntry) {
             FileHistory::createEntryForFile($this);
         }
 
-        $this->store->delete(null, ['file', FileHistory::VARIANT_PREFIX . '*']);
+        $store->delete(null, ['file', FileHistory::VARIANT_PREFIX . '*']);
+    }
+
+    /**
+     * @param $insert
+     * @param $changedAttributes
+     *
+     * @throws ErrorException
+     */
+    public function afterSave($insert, $changedAttributes)
+    {
+        parent::afterSave($insert, $changedAttributes);
+
+        /**
+         *  Make sure the unique index ux-file-object is not violated. Hence, if sort_order is zero (0) set it to the PK.
+         *
+         * @see self::updateAttributes()
+         */
+        $this->updateAttributes([]);
     }
 
     /**
      * Steps that must be performed after a new file content has been set.
      */
     private function afterNewStoredFile()
     {
-        if ($this->store->has()) {
+        $store = $this->getStore();
+
+        if ($store->has()) {
             // Make sure to update updated_by & updated_at and avoid save()
             $this->beforeSave(false);
 
+            $filename = $store->get();
             $this->updateAttributes([
-                'hash_sha1' => sha1_file($this->store->get()),
-                'size' => filesize($this->store->get()),
+                'hash_sha1' => sha1_file($filename),
+                'size' => filesize($filename),
                 'updated_by' => $this->updated_by,
                 'updated_at' => $this->updated_at,
             ]);
             $this->trigger(self::EVENT_AFTER_NEW_STORED_FILE);
         }
     }
+
+        /**
+         * Make sure the unique index ux-file-object is not violated. Hence, if sort_order is zero (0) set it to the PK.
+         *
+         * @param array|ArrayAccess $attributes
+         *
+         * @return void
+         * @throws ErrorException
+         * @see self::afterSave()
+         */
+    public function updateAttributes($attributes): void
```

Comment:
I would remove this method. At least for now.

Currently in HumHub and Yii2  this method is quite low level and never executes any validations or events.

I'm unsure if we should start doing that or leave it as it is.